neps/nep-0461.md

```diff
@@ -0,0 +1,72 @@
+| $2^{31}$ + NEP-number  | Off-chain Message, e.g. [`signMessage`](https://github.com/near/NEPs/pull/413) |
+
+## Motivation
+As the ecosystem grows, more standards will appear, enabling users to sign messages that are not transactions (e.g. [NEP413](https://github.com/near/NEPs/pull/413/), and [NEP366](https://github.com/near/NEPs/blob/master/neps/nep-0366.md)). If such messages collide with transactions, a malicious app could ask users to sign them, only to later relay them as valid transactions to the network.
+
+## Rationale and alternatives
+
+In NEAR, users sign transactions that are later relayed to the network. At the moment of being relayed, the transactions are simple streams of bytes that a node then parses into a [valid transaction](https://nomicon.io/RuntimeSpec/Transactions).
+
+As the protocol expands, more messages will be added for users to sign. We want users to be able to use the protocol safely, making sure that all those signed message cannot be interpreted as a valid transaction.
+
+Including a `u32` fixed prefix will enable to easily distinguish transactions from messages, while also ensuring that they never overlap. This is because the first parameter of a [valid transactions](https://nomicon.io/RuntimeSpec/Transactions) is an `AccountId`, which is encoded as `string length <u32>` + `string chars <u32[]>`. Adding a prefix in the range of $2^30 - 2^32$ will automatically make the non-transactional message to deserialize into an invalid transaction, since no `AccountId` has more than 64 chars.
+
+## Specification
+
+### On-Chain Messages
+All actionable messages, i.e. those that ARE validated on-chain, should have a structure which first parameter is a `prefix: u32` with fixed value $2^{30} + NEP-number$.
+
+For example, if we implement a new actionable message, destinated to be used on-chain, and the proposal has the number `NEP400`, then the structure must start with the prefix `1073742224`:
+
+```ts
+Structure{
+    prefix: u32 = 1073742224 // prefix fixed to 2**30 + 400 == 1073742224
+    // Rest of the attributes go here
+}
+```
+
+### Off-Chain Messages
+All non-actionable messages, i.e. those that are NOT validated on-chain, should have a structure which first parameter is a `prefix: u32` with fixed value $2^{31} + NEP-number$.
+
+For example, if we implement a new non-actionable message, destinated to be used off-chain, and the proposal has the number `NEP400`, then the structure must start with the prefix `2147484048`:
+
+```ts
+Structure{
+    prefix: u32 = 2147484048 // prefix fixed to 2**31 + 400 == 2147484048
+    // Rest of the attributes go here
+}
+```
+
+## Security Implications
+
+By implementing this NEP, we can be sure that all new messages do not collide with classic transactions.
+
+## Drawbacks
+
+A protocol change would be needed to ensure that `actionable messages` are treated as such, and not be confused with classic transactions. 
```

Comment:
I don't think that is true. The point of picking these high numbers is to avoid any potential conflict with existing transactions.

Reply:
removed the sentence